src/serac/numerics/functional/tests/quadrature_data_cuda.cu

```diff
@@ -275,8 +275,8 @@ struct state_manager_varying_qfunction {
     mutated_data[idx++] = state;
     return u;
   }
-  DeviceArray<typename wrapper_t::value_type>& mutated_data;
-  int                                          idx = 0;
+  axom::Array<typename wrapper_t::value_type, 1, axom::MemorySpace::Unified>& mutated_data;
```

Comment:
Is there a reason you didnt just use the ~~`DynamicArray`~~ `UnifiedArray` alias here and below?